modules/http/src/main/java/org/mule/module/http/internal/listener/grizzly/MuleSslFilter.java

```diff
@@ -35,10 +39,18 @@ public MuleSslFilter(SSLEngineConfigurator serverSSLEngineConfigurator, SSLEngin
     @Override
     public NextAction handleRead(FilterChainContext ctx) throws IOException
     {
-        ctx.getAttributes().setAttribute(HTTPS.getScheme(), true);
-        NextAction nextAction = super.handleRead(ctx);
-        ctx.getAttributes().setAttribute(SSL_SESSION_ATTRIBUTE_KEY, getSslSession(ctx));
-        return nextAction;
+        try
+        {
+            ctx.getAttributes().setAttribute(HTTPS.getScheme(), true);
+            NextAction nextAction = super.handleRead(ctx);
+            ctx.getAttributes().setAttribute(SSL_SESSION_ATTRIBUTE_KEY, getSslSession(ctx));
+            return nextAction;
+        }
+        catch (SSLHandshakeException e)
+        {
+            logger.error("SSL handshake error: " + e.getMessage());
```

Comment:
Doesn't this cause error to be logged twice, once here and once when it gets to exception handler?